src/test/java/io/xstefank/wildfly/bot/PREditTest.java

```diff
@@ -49,37 +54,48 @@ void configFileNullTest() throws IOException {
 
         GitHubAppTesting.given()
             .github(mocks -> mocks.configFileFromString("wildfly-bot.yml", wildflyConfigFile))
-            .when().payloadFromClasspath("/pr-success-checks.json")
+            .when().payloadFromString(gitHubJson.jsonString())
             .event(GHEvent.PULL_REQUEST)
             .then().github(mocks -> {
                 GHRepository repo = mocks.repository("xstefank/wildfly");
-                Mockito.verify(repo).createCommitStatus("40dbbdde147294cd8b29df16d79fe874247d8053",
+                Mockito.verify(repo).createCommitStatus(gitHubJson.commitSHA(),
                     GHCommitState.SUCCESS, "", "\u2705 Correct", "Format");
             });
     }
 
     @Test
     void incorrectPRFailTest() throws IOException {
+        GitHubJson gitHubJson = GitHubJson.builder("pr-template.json")
+                .action(Action.EDITED)
+                .pullRequestTitle("Foobar")
+                .build();
+
         GitHubAppTesting.given()
             .github(mocks -> mocks.configFileFromString("wildfly-bot.yml", wildflyConfigFile))
-            .when().payloadFromClasspath("/pr-fail-checks.json")
+            .when().payloadFromString(gitHubJson.jsonString())
             .event(GHEvent.PULL_REQUEST)
             .then().github(mocks -> {
                 GHRepository repo = mocks.repository("xstefank/wildfly");
-                Mockito.verify(repo).createCommitStatus("860035425072e50c290561191e90edc90254f900",
+                Mockito.verify(repo).createCommitStatus(gitHubJson.commitSHA(),
                     GHCommitState.ERROR, "", "\u274C title-check: Wrong content of the title!", "Format");
             });
     }
 
     @Test
     void correctPRSuccessTest() throws IOException {
+        GitHubJson gitHubJson = GitHubJson.builder("pr-template.json")
+                .action(Action.EDITED)
+                .pullRequestTitle("WFLY-1234 Test")
+                .pullRequestBody("JIRA: https://issues.redhat.com/browse/WFLY-666")
+                .build();
```

Comment:
it could probably be just title and just body, because you are already parsing PR in the JSON. If we would introduce issues or similar in the future we would probably need a slightly different builder.